core/sched.c

```diff
@@ -100,21 +100,21 @@ static void _unschedule(thread_t *active_thread)
 #endif
 }
 
-int __attribute__((used)) sched_run(void)
+thread_t *__attribute__((used)) sched_run(void)
 {
+    sched_context_switch_request = 0;
     thread_t *active_thread = thread_get_active();
+    thread_t *previous_thread = active_thread;
 
-    if (!IS_USED(MODULE_CORE_IDLE_THREAD)) {
-        if (!runqueue_bitcache) {
-            if (active_thread) {
-                _unschedule(active_thread);
-                active_thread = NULL;
-            }
-
-            do {
-                sched_arch_idle();
-            } while (!runqueue_bitcache);
+    if (!IS_USED(MODULE_CORE_IDLE_THREAD) && !runqueue_bitcache) {
+        if (active_thread) {
+            _unschedule(active_thread);
         }
+        active_thread = NULL;
```

Comment:
any reason for moving this out of the loop?

Reply:
Personal preference I think. The compiler doesn't care, binaries are the same between inside and outside the if statement.

Reply:
Then please move it back, original author personal preference has precedence! :)

unless you find it clearer this way, like, it is definitely null now?

Reply:
> unless you find it clearer this way, like, it is definitely null now?

I find this slightly more readable, I don't have to figure out that after the conditional the `active_thread` is always `NULL`. 

> Then please move it back, original author personal preference has precedence! :)

I'm fine with moving it back. Just let me know if you find the old version more readable.

Reply:
I actually do, thanks!

Reply:
Reverted!